cmake/modules/extensions.cmake

```diff
@@ -2492,8 +2494,14 @@ endfunction()
 # overridden by setting ZEPHYR_TOOLCHAIN_VARIANT directly in the CMake cache
 # using `-DZEPHYR_TOOLCHAIN_VARIANT=<val>`, then the value from the cache is
 # returned.
+#
+# There is an exclusion to the above rule when the MERGE option is used, in
+# which case the Sysbuild defined variable will be appended to the list last,
+# not first, this is so that local configuration to images will override the
+# default Sysbuild configuration, for example if a board directory is specified
+# on an image-level basis.
```

Comment:
I disagree on this.

Firstly, it's not just the sysbuild variables that is appended last, also any cache variables.

This means that if a user does `cmake -DFOO=bar <sample>` and `sample` has `set(FOO baz)`, then order becomes `baz;bar`, and thus `baz` has highest precedence when using `zephyr_get(FOO MERGE)`.
That prevents users from enforcing a specific value from having highest precedence.

In such case, the value provided by the user (in this case `bar`) should take precedence over the sample defined value (`baz`).

A use-case could is the `BOARD_ROOT` where user can specify extra roots on command line.
Today, BOARD_ROOT specified on command line take highest precedence but if `BOARD_ROOT` starts using `zephyr_get(BOARD_ROOT MERGE)` only on sysbuild, then that would suddenly be a behavioral change in the build system. 

If a user today does:
`west build ... <sample> -- -DBOARD_ROOT=<my_custom_root>`
then that user should experience the same behavior when building that same sample using sysbuild, that is:
`west build ... --sysbuild <sample> -- -DBOARD_ROOT=<my_custom_root>`

With this PR, the order will be different between the two runs.
First run will have `<my_custom_root>` first in list, second run will have it last in list.

Reply:
I tested this and the order changing does not happen, using `cmake -GNinja -DBOARD=nrf52840dk_nrf52840 -DAPP_DIR=.. -DSB_CONFIG_BOOTLOADER_MCUBOOT=y -Dblob_BOARD_ROOT=/tmp/cc/example-application -DBOARD_ROOT=/tmp/bb/nrfxlib/ /tmp/aa/zephyr/share/sysbuild/` I correctly see `/tmp/bb/nrfxlib/` for sysbuild, `/tmp/bb/nrfxlib/;/tmp/aa/zephyr` for mcuboot, `/tmp/bb/nrfxlib/;/tmp/aa/zephyr` for the main app and `/tmp/cc/example-application;/tmp/bb/nrfxlib/;/tmp/aa/zephyr` for the blob target

Reply:
Have reworked this and made the order the same for both.